scylla/src/frame/cql_types_test.rs

```diff
@@ -96,4 +97,44 @@ async fn test_cql_types() {
             println!("float value: {}", bool_val);
         }
     }
+
+    // Varint
+
+    session
+        .query(
+            "CREATE TABLE IF NOT EXISTS ks.varint_table (value varint, id int PRIMARY KEY)",
+            &[],
+        )
+        .await
+        .unwrap();
+
+    let val_positive = 10000.to_bigint().unwrap();
+    let val_negative = (-10000).to_bigint().unwrap();
+
+    session
+        .query(
+            "INSERT INTO ks.varint_table (value, id) VALUES (?, 1)",
+            (val_positive,),
+        )
+        .await
+        .unwrap();
+
+    session
+        .query(
+            "INSERT INTO ks.varint_table (value, id) VALUES (?, 1)",
+            (val_negative,),
+        )
+        .await
+        .unwrap();
+
+    if let Some(rows) = session
+        .query("SELECT value FROM ks.varint_table", &[])
+        .await
+        .unwrap()
+    {
+        for row in rows.into_typed::<(BigInt,)>() {
+            let bool_val: BigInt = row.unwrap().0;
+            println!("varint value: {}", bool_val);
```

Comment:
Let's validate the output with an assert instead of just printing it, and try some more tricky values, e.g. `11111111111111111111111111111111111111111111111111111111111111111111111111111111111`.

Also, let's add a test case for decimals as well - including values with decimal part, e.g. `BigDecimal::from_str(&"-123.456789999999999")`.

Please also add some tests which expect failures - e.g. decimals with incorrect scale.

Reply:
I'd also suggest adding a test which inserts a value directly encoded in a CQL string (e.g. `INSERT INTO ks.varint_table (value, id) VALUES (123456789123456789123456789123456789, 2)`, then reads it and verifies the value.

With more complex data types it's easy to make a mistake where encoding a rust value gives a wrong value, but deserializing it back gives good result (e.g. imagine if serialization and deserialization accidentally flipped the sign of the number). Such test case should protect against it.

Reply:
Ideally we should do the test I mentioned with a few different values - at least, it would be nice to test `0` and some negative number.

Reply:
I just added some more testcases - in cql_types_test and unit tests in result.rs for deserializers, using the specification of BigDecimal.